### PR TITLE
Ensdf processing bricc

### PR DESCRIPTION
**Pull request by howland**
Adds Bricc, Bldhst & Radlist ensdf evaluation tools, and a tutorial on how to use them (and the other ensdf evaluation tools).

### CODE REVIEW

**Comment by howland** [comment]
Note: Bricc interface is not currently working, please ignore any BRICC related sections of the PR for now.  I'll have them fixed shortly.

File: pyne/ensdf_processing.py [hunk 1]
```diff
@@ -79,6 +80,97 @@ def alphad(inputdict_unchecked):
...
+
+    Returns
+    -------
+    rtn : dictionary
+        Everything in input dictionary is returned if BRICC completes successfully.
+        Additional dictionary entries including:
+            output_file_directory : string, the directory all produced bricc output files will be
+                                    located.
+            bricc_output : string, Only for interactive use: data printed to command line.
+    NOTE:
```
Comment (scopatz) [inline pyne/ensdf_processing.py]: In numpydoc, this should be 

```
Notes
-----
my note
```

File: pyne/ensdf_processing.py [hunk 2]
```diff
@@ -79,6 +80,97 @@ def alphad(inputdict_unchecked):
...
+    verify_download_exe(compressed_exe_path, bricc_url, compressed = True, decomp_path = decomp_exe_path, dl_size = 127232)
+
+    # check if BriIccHome environment variable has been set (needed by BRICC executable)
+    if not os.environ.get('BrIccHome'):
+        os.environ['BrIccHome'] = str(exe_dir)
+
+    input_type = inputdict_unchecked['input_type']
+    output_dict = inputdict_unchecked
+
+    if(input_type == 'interactive'):
```
Comment (scopatz) [inline pyne/ensdf_processing.py]: no need for parentheses here.  This is not C or C++ :)

File: tests/test_ensdf_processing.py [hunk 3]
```diff
@@ -163,8 +226,8 @@ def file_comp(file_out, file_ref, exceptions):
         type 4: carriage return vs. non standard return type.
             options: line number of return.
     '''
-    f_out = open(file_out, 'r')
-    f_ref = open(file_ref, 'r')
+    f_out = open(file_out, 'r',encoding="latin-1")
+    f_ref = open(file_ref, 'r',encoding="latin-1")
```
Comment (scopatz) [inline tests/test_ensdf_processing.py]: It is not clear why you need this encoding.  Wouldn't utf8 or ascii be more applicable?

Comment (howland) [inline tests/test_ensdf_processing.py]: @scopatz I'm pretty sure the bldhst analysis program writes its output with some other encoding.  When utf8 or ascii are used, a couple of invalid continuation byte errors occur when trying to read in the file for comparison.  The only encoding I could get to work with bldhst was 'latin-1.'

File: pyne/ensdf_processing.py (hunk 1 above)
Comment (howland) [inline pyne/ensdf_processing.py]: Fixed!

File: pyne/ensdf_processing.py (hunk 2 above)
Comment (howland) [inline pyne/ensdf_processing.py]: Right.  Fixed!!

**Comment by howland** [comment]
@scopatz / anybody Do you know if there is any way to increase the timeout time for a single nose test  on travis-ci?  

The default timeout time is 10 according to the docs, which isn't nearly long enough for two of my new tests.  Both tests prompt the executable (for the analysis program they are attempting to use) to be downloaded, as source code wasn't available.

If not, should I remove these tests, leaving two of the analysis programs without test?

**Comment by makeclean** [comment]
travis_wait

**Comment by howland** [comment]
Ah ok cool.  I also missed that each test is run with a separate nose call, I should also be able to change nose's timeout time. 

```
  echo "Testing $x:" 
  echo 
  nosetests "$x"
```

File: pyne/ensdf_processing.py [hunk 4]
```diff
@@ -5,40 +5,82 @@
...
+
+def setup_bricc():
+    exe_path = path_to_exe('bricc')
+    exe_dir = path_to_exe('')[:-1]
+    compressed_exe_path = exe_path + '.tar.gz'
+
+    bricc_url = "http://www.nndc.bnl.gov/nndcscr/ensdf_pgm/analysis/BrIcc/Linux/BriccV23-Linux.tgz"
+    decomp_exe_path = path_to_exe('')
+    decomp_options = ['bricc', '.tgz', True]
+    download_exe(compressed_exe_path, bricc_url, compressed = True, decomp_path = decomp_exe_path, dl_size = 127232)
```
Comment (kellyrowland) [inline pyne/ensdf_processing.py]: This line is over the 80-char limit.

File: pyne/ensdf_processing.py [hunk 5]
```diff
@@ -5,40 +5,82 @@
...
+    setup_gabs()
+    print("Downloading RADLIST executable")
+    setup_radlist()
+
+def setup_bricc():
+    exe_path = path_to_exe('bricc')
+    exe_dir = path_to_exe('')[:-1]
+    compressed_exe_path = exe_path + '.tar.gz'
+
+    bricc_url = "http://www.nndc.bnl.gov/nndcscr/ensdf_pgm/analysis/BrIcc/Linux/BriccV23-Linux.tgz"
```
Comment (kellyrowland) [inline pyne/ensdf_processing.py]: This line is over the 80-char limit, but I'm not sure how to handle that stylistically given that the URL itself is 81 chars... @scopatz ?

File: pyne/ensdf_processing.py [hunk 6]
```diff
@@ -79,6 +121,113 @@ def alphad(inputdict_unchecked):
...
+        dictionary that must have the following key-pair values:
+            xx
+            xx
+
+    Returns
+    -------
+    rtn : dictionary
+        Everything in input dictionary is returned if BRICC completes successfully.
+        Additional dictionary entries including:
+            output_file_directory : string, the directory all produced bricc output files will be
```
Comment (kellyrowland) [inline pyne/ensdf_processing.py]: more >=80-char lines in here.

File: tests/test_ensdf_processing.py [hunk 7]
```diff
@@ -209,3 +271,12 @@ def file_comp(file_out, file_ref, exceptions):
     logft = test_logft()
     radd = test_radd()
     ruler = test_ruler()
+
+    ### The following code runs the additional setup code and runs tests
+    ### for each.  They are not included in the main test suit because
```
Comment (kellyrowland) [inline tests/test_ensdf_processing.py]: "suit" should be "suite"

**Comment by kellyrowland** [comment]
This looks good to me overall, just needs some cleanup in terms of shortening lines to be <= 80 chars.

File: pyne/ensdf_processing.py (hunk 5 above)
Comment (gonuke) [inline pyne/ensdf_processing.py]: Can split the string over two lines with concatenation.....?

**Comment by howland** [comment]
@kellyrowland Thanks!  I revised those lines and a handful of others.

**Comment by kellyrowland** [comment]
Looks good, merging.

**Pull request merged**